From 00ae68caf9538556c30a5617d45ecb58d465e7de Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Thu, 17 Oct 2024 11:09:03 +0200
Subject: [PATCH 1330/1695] dts: stm32: handle orange LED on stm32mp215f-dk

Handle the orange LED driven by PMIC GPO3 on stm32mp215f-dk.
The LED has to be ON in shutdown, OFF else.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b8332c3edec398b9af02895c3e547a57ac900d49 dts: stm32: handle orange LED on stm32mp215f-dk]
Change-Id: I844091e9c16d8413c4bc3f359628954fb19268a1
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/411990
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index a29b3f4bf..7b0a92c23 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -242,6 +242,27 @@
 					regulator-off-in-suspend;
 				};
 			};
+			v_gpo3: gpo3 {
+				regulator-name = "gpo3";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+				st,pwrctrl-sel = <1>;
+				st,pwrctrl-enable;
+
+				default {
+					regulator-on-in-suspend;
+				};
+				lplv {
+					regulator-on-in-suspend;
+				};
+				standby {
+					regulator-on-in-suspend;
+				};
+				off {
+					regulator-off-in-suspend;
+				};
+			};
 		};
 	};
 };
-- 
2.47.2

